### PR TITLE
Package mpg123.1.0

### PR DESCRIPTION
**Pull request by mbacarella**
### `mpg123.1.0`
MP3 decoding library
OCaml bindings to libmpg123

---
* Homepage: https://github.com/mbacarella/mpg123
* Source repo: git+https://github.com/mbacarella/mpg123.git
* Bug tracker: https://github.com/mbacarella/mpg123/issues

---
:camel: Pull-request generated by opam-publish v2.2.0

### CODE REVIEW

File: packages/mpg123/mpg123.1.0/opam [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+license: "LGPL-2.1-or-later WITH OCaml-LGPL-linking-exception"
+homepage: "https://github.com/mbacarella/mpg123"
+bug-reports: "https://github.com/mbacarella/mpg123/issues"
+depends: [
+  "dune" {>= "3.2"}
+  "ctypes" {>= "0.13.0"}
+  "ocaml" {>= "4.03"}
+  "ppx_inline_test"
+  "odoc" {with-doc}
+]
```
Comment (mseri) [inline packages/mpg123/mpg123.1.0/opam:16]:
```suggestion
  "conf-automake"
]
```
Due to
```
# Next type 'make' and then 'make install'.
# File "dune", line 3, characters 0-753:
#  3 | (rule
#  4 |  (targets libmpg123.a dllmpg123%{ext_dll})
#  5 |  (deps
# ....
# 21 |     (copy
# 22 |      vendor/mpg123-1.29.2/src/libmpg123/.libs/libmpg123%{ext_dll}
# 23 |      dllmpg123%{ext_dll})))))
# (cd _build/default/vendor/mpg123-1.29.2 && /bin/bash -e -u -o pipefail -c make)
# CDPATH="${ZSH_VERSION+.}:" && cd . && /bin/bash '/home/opam/.opam/4.04/.opam-switch/build/mpg123.1.0/_build/default/vendor/mpg123-1.29.2/build/missing' aclocal-1.16 -I m4
# /home/opam/.opam/4.04/.opam-switch/build/mpg123.1.0/_build/default/vendor/mpg123-1.29.2/build/missing: line 81: aclocal-1.16: command not found
# WARNING: 'aclocal-1.16' is missing on your system.
#          You should only need it if you modified 'acinclude.m4' or
#          'configure.ac' or m4 files included by 'configure.ac'.
#          The 'aclocal' program is part of the GNU Automake package:
#          <https://www.gnu.org/software/automake>
#          It also requires GNU Autoconf, GNU m4 and Perl in order to run:
#          <https://www.gnu.org/software/autoconf>
#          <https://www.gnu.org/software/m4/>
#          <https://www.perl.org/>
# make: *** [Makefile:2944: aclocal.m4] Error 127
```

**Comment by dinosaure** [comment]
Currently, the `configure` of the vendored `libmpg123` fails with:
```
#=== ERROR while compiling mpg123.1.0 =========================================#
# context              2.2.0~alpha~dev | linux/x86_64 | ocaml-base-compiler.4.04.2 | pinned(https://github.com/mbacarella/mpg123/archive/refs/tags/1.0.tar.gz)
# path                 ~/.opam/4.04/.opam-switch/build/mpg123.1.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p mpg123 -j 47 @install
# exit-code            1
# env-file             ~/.opam/log/mpg123-7-786157.env
# output-file          ~/.opam/log/mpg123-7-786157.out
### output ###
# (cd _build/default/vendor/mpg123-1.29.2 && /bin/bash -e -u -o pipefail -c 'sh configure --with-pic=yes --with-audio=dummy --enable-static=yes --enable-silent-rules')
# checking build system type... x86_64-pc-linux-gnu
# checking host system type... x86_64-pc-linux-gnu
# checking for a BSD-compatible install... /usr/bin/install -c
# checking whether build environment is sane... yes
# checking for a thread-safe mkdir -p... /bin/mkdir -p
# checking for gawk... no
# checking for mawk... mawk
# checking whether make sets $(MAKE)... yes
# checking whether make supports nested variables... yes
# checking for -Werror in CFLAGS (It breaks tests)... no
# checking if you are up to something totally static with LDFLAGS/CFLAGS... no
# checking whether make supports the include directive... yes (GNU style)
# checking for gcc... gcc
# checking whether the C compiler works... yes
# checking for C compiler default output file name... a.out
# checking for suffix of executables... 
# checking whether we are cross compiling... no
# checking for suffix of object files... o
# checking whether we are using the GNU C compiler... yes
# checking whether gcc accepts -g... yes
# checking for gcc option to accept ISO C89... none needed
# checking whether gcc understands -c and -o together... yes
# checking dependency style of gcc... gcc3
# checking dependency style of gcc... gcc3
# checking for gcc option to accept ISO C99... none needed
# checking how to run the C preprocessor... gcc -E
# checking for cos in -lm... yes
# checking for grep that handles long lines and -e... /bin/grep
# checking for egrep... /bin/grep -E
# checking for an ANSI C-conforming const... yes
# checking for inline... inline
# checking for ANSI C header files... yes
# checking for sys/types.h... yes
# checking for sys/stat.h... yes
# checking for stdlib.h... yes
# checking for string.h... yes
# checking for memory.h... yes
# checking for strings.h... yes
# checking for inttypes.h... yes
# checking for stdint.h... yes
# checking for unistd.h... yes
# checking whether byte ordering is bigendian... no
# checking windows.h usability... no
# checking windows.h presence... no
# checking for windows.h... no
# checking for library containing dlopen... -ldl
# checking for library containing dlsym... none required
# checking for library containing dlclose... none required
# checking dlfcn.h usability... yes
# checking dlfcn.h presence... yes
# checking for dlfcn.h... yes
# checking for dlopen... yes
# checking for dlsym... yes
# checking for dlclose... yes
# We found a runtime loader: Modules enabled.
# checking how to print strings... printf
# checking for a sed that does not truncate output... /bin/sed
# checking for fgrep... /bin/grep -F
# checking for ld used by gcc... /usr/bin/ld
# checking if the linker (/usr/bin/ld) is GNU ld... yes
# checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
# checking the name lister (/usr/bin/nm -B) interface... BSD nm
# checking whether ln -s works... yes
# checking the maximum length of command line arguments... 1572864
# checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
# checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
# checking for /usr/bin/ld option to reload object files... -r
# checking for objdump... objdump
# checking how to recognize dependent libraries... pass_all
# checking for dlltool... no
# checking how to associate runtime and link libraries... printf %s\n
# checking for ar... ar
# checking for archiver @FILE support... @
# checking for strip... strip
# checking for ranlib... ranlib
# checking command to parse /usr/bin/nm -B output from gcc object... ok
# checking for sysroot... no
# checking for a working dd... /bin/dd
# checking how to truncate binary pipes... /bin/dd bs=4096 count=1
# configure: line 8672: /usr/bin/file: No such file or directory
# checking for mt... no
# checking if : is a manifest tool... no
# checking for dlfcn.h... (cached) yes
# checking for objdir... .libs
# checking if gcc supports -fno-rtti -fno-exceptions... no
# checking for gcc option to produce PIC... -fPIC -DPIC
# checking if gcc PIC flag -fPIC -DPIC works... yes
# checking if gcc static flag -static works... yes
# checking if gcc supports -c -o file.o... yes
# checking if gcc supports -c -o file.o... (cached) yes
# checking whether the gcc linker (/usr/bin/ld) supports shared libraries... yes
# checking whether -lc should be explicitly linked in... no
# checking dynamic linker characteristics... GNU/Linux ld.so
# checking how to hardcode library paths into programs... immediate
# checking whether stripping libraries is possible... yes
# checking if libtool supports shared libraries... yes
# checking whether to build shared libraries... yes
# checking whether to build static libraries... yes
# checking what extension is used for runtime loadable modules... .so
# We assume IEEE754 floating point format.
# checking if .balign is present... yes
# checking __attribute__((aligned(16)))... yes
# checking if assembler supports AVX instructions... yes
# checking stdio.h usability... yes
# checking stdio.h presence... yes
# checking for stdio.h... yes
# checking for stdlib.h... (cached) yes
# checking for string.h... (cached) yes
# checking for unistd.h... (cached) yes
# checking sched.h usability... yes
# checking sched.h presence... yes
# checking for sched.h... yes
# checking sys/ioctl.h usability... yes
# checking sys/ioctl.h presence... yes
# checking for sys/ioctl.h... yes
# checking for sys/types.h... (cached) yes
# checking for stdint.h... (cached) yes
# checking limits.h usability... yes
# checking limits.h presence... yes
# checking for limits.h... yes
# checking for inttypes.h... (cached) yes
# checking sys/time.h usability... yes
# checking sys/time.h presence... yes
# checking for sys/time.h... yes
# checking sys/wait.h usability... yes
# checking sys/wait.h presence... yes
# checking for sys/wait.h... yes
# checking sys/resource.h usability... yes
# checking sys/resource.h presence... yes
# checking for sys/resource.h... yes
# checking sys/signal.h usability... yes
# checking sys/signal.h presence... yes
# checking for sys/signal.h... yes
# checking signal.h usability... yes
# checking signal.h presence... yes
# checking for signal.h... yes
# checking sys/select.h usability... yes
# checking sys/select.h presence... yes
# checking for sys/select.h... yes
# checking dirent.h usability... yes
# checking dirent.h presence... yes
# checking for dirent.h... yes
# checking for sys/stat.h... (cached) yes
# checking for special C compiler options needed for large files... no
# checking for _FILE_OFFSET_BITS value needed for large files... no
# checking for size_t... yes
# checking for uintptr_t... yes
# checking for ssize_t... yes
# checking for off_t... yes
# checking for int32_t... yes
# checking for int64_t... yes
# checking for uint32_t... yes
# checking for int16_t... yes
# checking for uint16_t... yes
# checking size of size_t... 8
# checking size of ssize_t... 8
# checking size of off_t... 8
# checking size of off64_t... 0
# checking size of int32_t... 4
# checking size of long... 8
# checking if we want to enable alias wrappers for largefile... yes
# checking for mmap... yes
# checking POSIX termios... yes
# checking for random... yes
# checking for sched_setscheduler... yes
# checking for setuid... yes
# checking for getuid... yes
# checking for setpriority... yes
# checking for strerror... yes
# checking for setlocale... yes
# checking for nl_langinfo... yes
# checking for mbstowcs... yes
# checking for wcstombs... yes
# checking for wcswidth... yes
# checking for iswprint... yes
# checking for atoll... yes
# checking for mkfifo... yes
# checking locale.h usability... yes
# checking locale.h presence... yes
# checking for locale.h... yes
# checking langinfo.h usability... yes
# checking langinfo.h presence... yes
# checking for langinfo.h... yes
# checking wchar.h usability... yes
# checking wchar.h presence... yes
# checking for wchar.h... yes
# checking wctype.h usability... yes
# checking wctype.h presence... yes
# checking for wctype.h... yes
# checking netdb.h usability... yes
# checking netdb.h presence... yes
# checking for netdb.h... yes
# checking sys/param.h usability... yes
# checking sys/param.h presence... yes
# checking for sys/param.h... yes
# checking sys/socket.h usability... yes
# checking sys/socket.h presence... yes
# checking for sys/socket.h... yes
# checking netinet/in.h usability... yes
# checking netinet/in.h presence... yes
# checking for netinet/in.h... yes
# checking arpa/inet.h usability... yes
# checking arpa/inet.h presence... yes
# checking for arpa/inet.h... yes
# checking for library containing gethostbyname... none required
# checking for library containing socket... none required
# checking for getaddrinfo... yes
# checking for sqrt in -lm... yes
# checking for powf in -lmx... no
# checking byteswap.h usability... yes
# checking byteswap.h presence... yes
# checking for byteswap.h... yes
# checking os2.h usability... no
# checking os2.h presence... no
# checking for os2.h... no
# checking for tcsetattr in -los2term... no
# checking if we want feature report function... yes
# Limiting outputs to build according to your preference: dummy dummy
# checking for pkg-config... no
# checking if you are too dumbing dumb for the dummy... no
# checking if you are too dumbing dumb for the dummy... no
# configure: line 18769: test: dummy: unary operator expected
# configure: line 18777: test: tinyalsa: unary operator expected
# configure: line 18785: test: alsa: unary operator expected
# configure: line 18793: test: qsa: unary operator expected
# configure: line 18801: test: coreaudio: unary operator expected
# configure: line 18809: test: esd: unary operator expected
# configure: line 18817: test: jack: unary operator expected
# configure: line 18825: test: nas: unary operator expected
# configure: line 18833: test: oss: unary operator expected
# configure: line 18841: test: portaudio: unary operator expected
# configure: line 18849: test: pulse: unary operator expected
# configure: line 18857: test: sdl: unary operator expected
# configure: line 18865: test: sndio: unary operator expected
# configure: line 18873: test: sun: unary operator expected
# configure: line 18881: test: win32: unary operator expected
# configure: line 18889: test: win32_wasapi: unary operator expected
# configure: line 18897: test: aix: unary operator expected
# configure: line 18905: test: alib: unary operator expected
# configure: line 18913: test: arts: unary operator expected
# configure: line 18921: test: hp: unary operator expected
# configure: line 18929: test: os2: unary operator expected
# configure: line 18937: test: sgi: unary operator expected
# configure: line 18945: test: mint: unary operator expected
# configure: line 18953: test: openal: unary operator expected
# checking if we have EOVERFLOW macro... yes
# checking that generated files are newer than configure... done
# configure: creating ./config.status
# config.status: creating Makefile
# config.status: creating libmpg123.pc
# config.status: creating libout123.pc
# config.status: creating libsyn123.pc
# config.status: creating mpg123.spec
# config.status: creating src/libmpg123/mpg123.h
# config.status: creating src/libout123/out123.h
# config.status: creating src/libsyn123/syn123.h
# config.status: creating src/config.h
# config.status: executing depfiles commands
# config.status: executing libtool commands
# 
#   mpg123 1.29.2
# 
#   Install path ............ /usr/local
#   CPU Optimization ........ x86-64
#   Compiler Optimization ... 2
#   Gapless Support ......... enabled
#   Debugging ............... disabled
#   Extreme debugging ....... disabled
#   Seek table size ......... 1000
#   FIFO support ............ enabled
#   Buffer .................. enabled
#   Network (http streams) .. enabled
#   Network Sockets ......... Posix
#   IPv6 (getaddrinfo) ...... enabled
#   File offsets ............ default
#   LFS alias symbols ....... enabled (64)
#   LFS alias type .......... off_t
#   Use yasm (for AVX only) . disabled
# 
#   Core libmpg123 features:
#   Layer I ................. enabled
#   Layer II ................ enabled
#   Layer III ............... enabled
#   NtoM resampling ......... enabled
#   downsampled decoding .... enabled
#   Feeder/buffered input ... enabled
#   ID3v2 parsing ........... enabled
#   String API .............. enabled
#   ICY parsing/conversion .. enabled
#   Error/warning messages .. enabled
#   Win32 Unicode File Open.. unneeded
#   Feature Report Function.. enabled
#   Output formats (nofpu will disable all but 16 or 8 bit!):
#   8 bit integer ........... enabled
#   16 bit integer .......... enabled
#   32/24 bit integer ....... enabled
#   real (32 bit float) ..... enabled
#   Equalizer ............... enabled
#   Optimization detail:
#   Integer conversion ...... quality
#   IEEE 754 hackery ........ enabled
#   New/old WRITE_SAMPLE .... disabled
#   new Huffman scheme ...... enabled
#   runtime tables .......... disabled
# 
# Note: Disabling core features is not commonly done and some combinations might
#       not build/work. If you encounter such a case, help yourself (and provide
#       a patch) or just poke the maintainers.
# 
#   libsyn123 special cases . enabled
# 
#   Modules ................. enabled
#   Checked audio modules ... dummy dummy
#   Detected audio support .. dummy dummy
#   Runtime order ........... 
# 
#   CPPFLAGS=' -DOPT_MULTI -DOPT_X86_64 -DOPT_GENERIC -DOPT_GENERIC_DITHER -DREAL_IS_FLOAT -DOPT_AVX '
#   CFLAGS='-O2 -fomit-frame-pointer -funroll-all-loops -finline-functions -ffast-math  -g -O2'
#   PROG_LIBS='-lm '
#        (derived from LIBS, only used for end-user binaries and modules)
#   LIBDL='-ldl'
#   LIBMPG123_LIBS='-lm '
#   LIBSYN123_LIBS='-lm'
#   LIBOUT123_LIBS='-lm '
# 
# Next type 'make' and then 'make install'.
# File "dune", line 3, characters 0-753:
#  3 | (rule
#  4 |  (targets libmpg123.a dllmpg123%{ext_dll})
#  5 |  (deps
# ....
# 21 |     (copy
# 22 |      vendor/mpg123-1.29.2/src/libmpg123/.libs/libmpg123%{ext_dll}
# 23 |      dllmpg123%{ext_dll})))))
# (cd _build/default/vendor/mpg123-1.29.2 && /bin/bash -e -u -o pipefail -c make)
# CDPATH="${ZSH_VERSION+.}:" && cd . && /bin/bash '/home/opam/.opam/4.04/.opam-switch/build/mpg123.1.0/_build/default/vendor/mpg123-1.29.2/build/missing' aclocal-1.16 -I m4
#  cd . && /bin/bash /home/opam/.opam/4.04/.opam-switch/build/mpg123.1.0/_build/default/vendor/mpg123-1.29.2/build/missing automake-1.16 --gnu
# CDPATH="${ZSH_VERSION+.}:" && cd . && /bin/bash '/home/opam/.opam/4.04/.opam-switch/build/mpg123.1.0/_build/default/vendor/mpg123-1.29.2/build/missing' autoconf
# configure.ac:276: error: possibly undefined macro: LT_SYS_MODULE_EXT
#       If this token and others are legitimate, please use m4_pattern_allow.
#       See the Autoconf documentation.
# make: *** [Makefile:2942: configure] Error 1
```

**Comment by mbacarella** [comment]
thanks @dinosaure, let me see what's going on

**Comment by avsm** [comment]
Have you had a chance to look at this, @mbacarella?

**Comment by mbacarella** [comment]
@avsm sorry, snowed under. will just withdraw this for now.

**Pull request closed without merging**